client/rest/client.go

```diff
@@ -210,21 +219,65 @@ func (s *restClient) Send(req *http.Request) (*http.Response, error) {
 		}
 		req.Header.Set("Authorization", s.token.String())
 	}
-	return s.send(req)
+	return s.send(req, 0, initBackoff)
+}
+
+func copyBody(req *http.Request) ([]byte, error) {
+	var (
+		body []byte
+		err  error
+	)
+	if req.Body != nil {
+		body, err = io.ReadAll(req.Body)
+		setBody(req, body)
+	}
+	return body, err
+}
+
+func setBody(req *http.Request, body []byte) {
+	if body != nil {
+		req.Body = io.NopCloser(bytes.NewBuffer(body))
```

Comment:
After looking into this a bit, it seems like it might be a good idea to at least also update `req.ContentLength` after changing `req.Body` like this.

Reply:
I'm not sure it would be necessary since in theory it's always copying the original content, but may not be a bad idea to just sanity check the length here.